master/internal/resourcemanagers/agent_resource_manager.go

```diff
@@ -495,7 +495,11 @@ func (a *agentResourceManager) createResourcePoolSummary(
 		}
 	}
 
-	resourceSummary := ctx.Ask(a.pools[poolName], GetResourceSummary{}).Get().(ResourceSummary)
+	response := ctx.Ask(a.pools[poolName], GetResourceSummary{})
+	if response.Error() != nil {
+		return &resourcepoolv1.ResourcePool{}, err
```

Comment:
I would return nil rather than an empty pool
```suggestion
		return nil, err
```